backend/src/main/java/fk/prof/backend/aggregator/ProfileWorkInfo.java

```diff
@@ -0,0 +1,81 @@
+package fk.prof.backend.aggregator;
+
+import com.koloboke.collect.map.hash.HashObjIntMap;
+import com.koloboke.collect.map.hash.HashObjIntMaps;
+import recording.Recorder;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+/**
+ * Non getter methods in this class are not thread-safe
+ * But, this should not be a concern because instances of this class are maintained for every work id
+ * Updates to state associated with a work id happens in context of a request (single thread)
+ */
+public class ProfileWorkInfo {
+  private AggregationStatus status = AggregationStatus.SCHEDULED;
+  private LocalDateTime startedAt = null, endedAt = null;
+  private int samples = 0;
+
+  private final HashObjIntMap<String> traceCoverages = HashObjIntMaps.newUpdatableMap();
+  private final Map<String, Integer> traceCoveragesUnmodifiableView = Collections.unmodifiableMap(traceCoverages);
+
+  private final Set<Recorder.WorkType> associatedWorkTypes = new HashSet<>();
+  private final Set<Recorder.WorkType> associatedWorkTypesUnmodifiableView = Collections.unmodifiableSet(associatedWorkTypes);
+
+  public void addTrace(String traceName, int coveragePct) {
+    traceCoverages.put(traceName, coveragePct);
+  }
+
+  public void associateWorkType(Recorder.WorkType workType) {
+    this.associatedWorkTypes.add(workType);
+  }
+
+  /**
+   * Returns unmodifiable view of trace name -> trace coverage data associated with a work id
+   * @return lookup map for trace coverage using trace name
+   */
+  public Map<String, Integer> getTraceCoverages() {
+    return this.traceCoveragesUnmodifiableView;
+  }
+
+  /**
+   * Returns unmodifiable view of all work types associated with the workId
+   * @return set of associated work types
+   */
+  public Set<Recorder.WorkType> getAssociatedWorkTypes() {
+    return this.associatedWorkTypesUnmodifiableView;
+  }
+
+  public void setStatus(AggregationStatus aggregationStatus) {
+    status = aggregationStatus;
+  }
+
+  public AggregationStatus getStatus() {
+    return status;
+  }
+
+  public LocalDateTime getStartedAt() {
```

Comment:
We have way too many getters and setters. This is bad OO from encapsulation and abstraction PoV. 

Because its a new codebase, we should perhaps delete all setters and getters and try to re-do LLD without them.

Eg. in this case getter is used by a test to validate things. This can be achieved by implementing equals and assert equality by creating an instance that test expects it to be like, etc.

We should really think through all the getters and setters. IntelliJ produces warnings for such things, they are often indicative of design issues. Let us try to fix them.

![screenshot-20170117 231201](https://cloud.githubusercontent.com/assets/12777/22032443/7f70c066-dd0a-11e6-9b74-df5724f3cdcf.png)